src/server/game/Maps/Map.cpp

```diff
@@ -2442,6 +2442,14 @@ float Map::GetWaterLevel(float x, float y) const
 
 bool Map::isInLineOfSight(float x1, float y1, float z1, float x2, float y2, float z2, uint32 phasemask, LineOfSightChecks checks, VMAP::ModelIgnoreFlags ignoreFlags) const
 {
+    if (!sWorld->getBoolConfig(CONFIG_VMAP_BLIZZLIKE_PVP_LOS))
+    {
+        if (IsBattlegroundOrArena())
+        {
+            ignoreFlags = VMAP::ModelIgnoreFlags::Nothing;
```

Comment:
By default, `ignoreFlags` is set to `VMAP::ModelIgnoreFlags::Nothing` already.
```
ignoreFlags |= VMAP::ModelIgnoreFlags::M2
```

Reply:
If it's nothing by default, then why do treestumps give los without setting it to nothing? I don't understand it.

Reply:
Is this line redundant then? https://github.com/azerothcore/azerothcore-wotlk/pull/11692/files#diff-cacd2443c0f46ce614d849924bdb64dd81aa4d5068b131e1c71ecac518b1c3e1R2460

Reply:
I'm lost here.

> If you add VMAP::ModelIgnoreFlags::M2, that M2 model will be ignored and thus you can cast through M2 models. 

We don't want this. I just made the change you suggested, I can shoot through treestumps with it.